pkg/remediator/reconcile/reconciler.go

```diff
@@ -85,11 +90,21 @@ func (r *reconciler) Remediate(ctx context.Context, id core.ID, obj client.Objec
 	// Record duration, even if there's an error
 	metrics.RecordRemediateDuration(ctx, metrics.StatusTagKey(err), id.WithVersion(""), start)
 
-	// Record conflict, if there was one
-	if err != nil && err.Code() == syncerclient.ResourceConflictCode {
-		metrics.RecordResourceConflict(ctx, id.WithVersion(""), commit)
+	if err != nil {
+		switch err.Code() {
+		case syncerclient.ResourceConflictCode:
+			// Record conflict, if there was one
+			metrics.RecordResourceConflict(ctx, id.WithVersion(""), commit)
+		case status.FightErrorCode:
+			operation := objDiff.Operation(r.scope, r.syncName)
+			metrics.RecordResourceFight(ctx, string(operation), id.WithVersion(""))
+			r.fightHandler.AddFightError(id, err)
```

Comment:
Now that we have a FightHandler... would it make sense to also add a ConflictHandler? It's kinda weird we add one here but not the other.

Reply:
I have a local PR ready for the ConflictHandler. Will push out for review once this PR is merged.